### PR TITLE
The Utf8FrameValidator only handles its own protocol violation exception

### PR DESCRIPTION
**Pull request by amizurov**
Motivation:

Currently `Utf8FrameValidator` catches all `CorruptedWebSocketFrameException` and trying to close the channel if it's open. This is incorrect behavior because `WebsocketFrameDecoder` throws the same type of exception, but with a different status code and reason, and does not close the channel if the `closeOnProtocolViolation` property is set to false.

Modification:

Correct that the Utf8FrameValidator only handles its own protocol violation exception  and sends the CloseWebsocketFrame with the appropriate status code and reason. Pass the `closeOnProtocolViolation` property to the `Utf8FrameValidator`;

Result:
The Utf8FrameValidator only handle its own protocol violation exception and consolidate with decoder config.

### CODE REVIEW

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java [hunk 1]
```diff
@@ -15,18 +15,12 @@
...
 import io.netty.handler.codec.CorruptedFrameException;
 import org.junit.jupiter.api.Test;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertNotNull;
-import static org.junit.jupiter.api.Assertions.assertNull;
-import static org.junit.jupiter.api.Assertions.assertTrue;
-import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (normanmaurer) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java:23]: please revert... we don't use wildcard imports.

Comment (normanmaurer) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java:23]: please revert... we don't use wildcard imports.

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java [hunk 2]
```diff
@@ -40,24 +34,41 @@ public void testCorruptedFrameExceptionInCheck() {
...
+    @Test
+    void testNotCloseOnProtocolViolation() {
+        EmbeddedChannel channel = new EmbeddedChannel(new Utf8FrameValidator(false));
+        TextWebSocketFrame frame = new TextWebSocketFrame(Unpooled.copiedBuffer(new byte[]{-50}));
+        try {
+            channel.writeInbound(frame);
+            fail();
+        } catch (CorruptedFrameException expected) {
+            // expected exception
+        }
```
Comment (normanmaurer) [inline codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java:46]: please use `assertThrows`

**Comment by normanmaurer** [comment]
@amizurov yeah just add the override to see if the CI pass or add some "ignore" to the pom.xml. Sounds like a bug of the plugin

**Comment by normanmaurer** [comment]
@amizurov yeah just add the override to see if the CI pass or add some "ignore" to the pom.xml. Sounds like a bug of the plugin

**Review by normanmaurer — CHANGES_REQUESTED** [review]

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java (hunk 1 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java:23]: Reverted

Comment (amizurov) [reply codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java]: Reverted

File: codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java (hunk 2 above)
Comment (amizurov) [reply codec-http/src/test/java/io/netty/handler/codec/http/websocketx/WebSocketUtf8FrameValidatorTest.java]: Fixed

**Comment by normanmaurer** [comment]
@amizurov thanks a lot!

**Pull request merged**